portfolio/src/main/webapp/nicknameFactory.js

```diff
@@ -0,0 +1,31 @@
+
+/**
+ * Fetches and displays relevant nickname information pertaining to the user from the nickname 
+ * servlet.
+ */
+async function fetchAndDisplayNicknameInfo(){
+  const response = await fetch('/nickname');
+  const json = await response.json();
+  const url = json.url;
+  const loggedInStatus = json.loggedin;
+  const nickname = json.nickname;
+  displayLoginLogoutOrForm(nickname,url,loggedInStatus);
+  
+}
+
+/** 
+ * Displays the login link if not signed in, else displays the nickname form.
+ * {@code nickname} is the user's nickname, {@code url} is the URL to display, and 
+ * {@code loggedInStatus} is a boolean representing whether the user is logged in.
+ */
+function displayLoginLogoutOrForm(nickname,url,loggedInStatus){
+  if (loggedInStatus){  
+    document.getElementById("nickname-form").style.display="block";
+    document.getElementById("nickname-input").value = nickname;
+    document.getElementById("loginlogoutinfo").innerHTML = "Logout <a href=\"" + url + "\">here</a>.</p>"
```

Comment:
JavaScript strings can use a mix of `'` and `"` characters, which means you could avoid escaping the `"` character like this:

```
'Logout <a href="' + url + '">here</a>.</p>'
```

Also, semicolons are technically optional in JavaScript, the [style guide](https://google.github.io/styleguide/jsguide.html#formatting-semicolons-are-required) recommends using them, so I would recommend adding a semicolon at the end of this line. (Same for the similar line below.)